pandas/tseries/tdi.py

```diff
@@ -391,6 +391,17 @@ def f(x):
             result = result.astype('int64')
         return result
 
+    def total_seconds(self):
+        """ Total duration of each element expressed in seconds. """
+        values = self.asi8
+        hasnans = self.hasnans
+        result = 1e-9 * values
+        if hasnans:
```

Comment:
use ``result = self._maybe_mask_results(result)`` (you can remove the ``hasnans`` as well)